File: python/pset/sentimental-readability/readability.py
from math import ceil
text = input("Enter text: ")


def calc_letter():
    count = 0
    for letter in text:
        if letter.isalnum():
            count += 1
    return count


def calc_word():
    count = 0
    for letter in text:
        if letter == " ":
            count += 1
    return count


def calc_sentence():
    count = 0
    for letter in text:
        if letter == '.' or letter == '!' or letter == '?':
            count += 1
    return count + 1


letter_count = calc_letter()
word_count = calc_word()
sentence_count = calc_sentence()


def calc_grade():
    L = (letter_count / word_count) * 100
    S = (sentence_count / word_count) * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    return ceil(index)


if calc_grade() >= 16:
    print("Grade 16+")
elif calc_grade() < 1:
    print("Before Grade 1")
else:
    print(f"Grade {calc_grade()}")

File: c/week_two/problem 1/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>


int calc_letter(string word);
int calc_sentence(string word);
int calc_word(string word);
int calc_grade(int letters, int sentences, int words);

int main(void)
{
    string text = get_string("Enter text: ");
    int total_letter = calc_letter(text);
    int total_sentence = calc_sentence(text);
    int total_word = calc_word(text);
    int grade = calc_grade(total_letter, total_sentence, total_word);

    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if(grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else {
        printf("Grade %i\n", grade);
    }
}

int calc_letter(string word)
{
    int total = 0;
    int str_length = strlen(word);

    for(int i = 0; i < str_length; i++)
    {
        if(isalpha(word[i]))
        {
            total = total + 1;
        }
    }
    return total;
}

int calc_sentence (string word)
{
    int total = 0;
    int str_length = strlen(word);

    for(int i = 0; i < str_length; i++)
    {
        char s = word[i];
        if(s == '.' || s == '!' || s == '?')
        {
            total = total + 1;
        }
    }
    return total + 1;
}

int calc_word (string word)
{
    int total = 0;
    int str_length = strlen(word);
    for(int i = 0; i < str_length; i++)
    {
        char s = word[i];
        if(s == ' ')
        {
            total = total + 1;
        }
    }

    return total;
}

int calc_grade (int letters, int sentences, int words)
{
    float L = ((float)letters / (float)words) * 100;
    float S = ((float)sentences / (float)words) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int toInt = round(index);
    return toInt;
}